tudatBundle/pagmo2/include/pagmo/algorithms/giACOmo.hpp

```diff
@@ -198,132 +243,159 @@ class gi_aco_mo
+                            //that generation), whereas the last vector represents the worst.
+                            for (decltype(m_ker) i=0u; i<m_ker; ++i){
+                                SA[i]={penalties[sorted_penalties[i]], X[sort_list[i]], fit[sort_list[i]]};
 
                             }
 
-                        }
-                    }
-                }
 
-                if (gen==1) {
 
-                    std::vector<std::vector<double>> *Sol_Arch=&SA;
 
-                    //I initialize the solution archive (SA): in order to do this I store the vectors generated in the first generation
-                    //by taking into account their penalty values. In this way, the first vector in the SA (i.e., the first row in which
-                    //penalty value, variables, objective functions values, equality constraints values, inequality constraints values are
-                    //stored) represents the best one (i.e., the one that has the smallest penalty function value among the individuals of
-                    //that generation), whereas the last vector represents the worst.
-                    for (decltype(sort_list.size()) i=0u; i<sort_list.size(); ++i){
-                        *Sol_Arch[i]={penalties[sorted_penalties[i]], X[sort_list[i]], fit[sort_list[i]]};
-                    }
+                        }
+                        else{
 
+                            update_SA(pop, sorted_penalties, sort_list, SA);
 
+                        }
 
 
-                }
-                else{
 
-                    update_SA(pop, sorted_penalties, sort_list, SA);
+                        //3 - compute pheromone values
 
-                }
+                        std::vector <double> omega;
+                        std::vector <double> sigma;
+                        pheromone_computation(omega, sigma, pop, SA);
 
 
+                        //4 - use pheromone values to generate new ants, which will become the future generation's variables
+                        //here you have to define a probability density function and use a random number generator to produce
+                        //the new ants from it
 
-                //3 - compute pheromone values
+                        //I create the vector of vectors where I will store all the new ants (i.e., individuals) which will be generated
+                        std::vector < std::vector <double> > new_ants;
+                        generate_new_ants( omega, sigma, SA, n_con, new_ants );
 
-                std::vector <double> omega;
-                std::vector <double> sigma;
-                pheromone_computation(omega, sigma); //you still have to define the inputs to pass
+                    }
+                    else{
+                        //here you should generate a new first population, because the one you selected had not produced at least
+                        //'m_ker' feasible solutions
```

Comment:
Here I would like to generate a new initial population if the first one does not have at least m_ker feasible individuals (i.e., there are not m_ker individuals which satisfy the equality and inequality constraints within the accuracy value).

Shall I do it by returning a pagmo_throw in which I advice the user to change the population because it does not have enough feasible elements, or is there a smarter way to do it?

Reply:
There is no repair method that works across problems, so I would throw.

Reply:
Okay, thanks.